pkg/pbq/store/wal/segment.go

```diff
@@ -51,6 +52,10 @@ type WAL struct {
 	fp *os.File
 	// wOffset is the write offset as tracked by the writer
 	wOffset int64
+	// prevSyncedWOffset is the write offset that is already synced as tracked by the writer
+	prevSyncedWOffset int64
+	// prevSyncedTime is the time when the last sync was made
+	prevSyncedTime time.Duration
```

Comment:
why is time duration? shouldn't it be `time.Time`? Because of this, you are doing a lot of conversions later on.

Reply:
Yes. I will change this.

Reply:
I did that because we want the duration to be more than some x nanoseconds before syncing again. So, I felt that duration might be better.  But, even i feel that I am converting a lot of times because of this. I will update this. Thanks.